src/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\HttpClient;
+
+use Sentry\Client;
+use Sentry\Options;
+
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var Options
+     */
+    protected $options;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkIdentifier;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkVersion;
+
+    public function __construct(Options $options, string $sdkIdentifier, string $sdkVersion)
+    {
+        $this->options = $options;
+        $this->sdkIdentifier = $sdkIdentifier;
+        $this->sdkVersion = $sdkVersion;
+    }
+
+    public function sendRequest(string $requestData): Response
+    {
+        $dsn = $this->options->getDsn();
+        if (null === $dsn) {
+            throw new \RuntimeException('The DSN option must be set to use the HttpClient.');
+        }
+
+        $curlHandle = curl_init();
+        curl_setopt($curlHandle, \CURLOPT_URL, $dsn->getEnvelopeApiEndpointUrl());
+        curl_setopt($curlHandle, \CURLOPT_HTTPHEADER, $this->getRequestHeaders());
+        curl_setopt($curlHandle, \CURLOPT_USERAGENT, $this->sdkIdentifier . '/' . $this->sdkVersion);
+        curl_setopt($curlHandle, \CURLOPT_TIMEOUT, $this->options->getHttpTimeout());
+        curl_setopt($curlHandle, \CURLOPT_CONNECTTIMEOUT, $this->options->getHttpConnectTimeout());
+        curl_setopt($curlHandle, \CURLOPT_SSL_VERIFYPEER, true);
```

Comment:
What if I want to disable this? i.e. if I have a local, self-hosted, not exposed, not behind TLS Sentry server?

[edit] I saw the TODO below, this row is a duplicate.

Reply:
I added `http_ssl_verify_peer`. Your use case is the same one I had in mind.